otlp/traces.go

```diff
@@ -215,13 +218,31 @@ func BytesToTraceID(traceID []byte) string {
 		}
 	case traceIDShortLength: // 8 bytes
 		encoded = make([]byte, 16)
+	case traceIDCorruptedLength: // 24 bytes
```

Comment:
nit: I think we could refactor this so we don't mix an early return for a special case with ones that expect to fall through.

Reply:
The choice was between hoisting the hex conversion into the other cases vs early return. I chose to minimize lines of change, but I'm happy to do it the other way.